tests/farmer_harvester/test_farmer.py

```diff
@@ -619,6 +644,290 @@ def assert_pool_errors_24h() -> None:
+        signage_point_index=case.signage_point_index,
+        peak_height=uint32(1),
+    )
+    pos = ProofOfSpace(
+        challenge=case.plot_challenge,
+        pool_public_key=case.pool_public_key,
+        pool_contract_puzzle_hash=case.pool_contract_puzzle_hash,
+        plot_public_key=case.plot_public_key,
+        size=case.plot_size,
+        proof=case.proof,
+    )
+    new_pos = NewProofOfSpace(
+        challenge_hash=case.challenge_hash,
+        sp_hash=case.sp_hash,
+        plot_identifier=case.plot_identifier,
+        proof=pos,
+        signage_point_index=case.signage_point_index,
+    )
+    p2_singleton_puzzle_hash = case.pool_contract_puzzle_hash
+    farmer.constants = dataclasses.replace(DEFAULT_CONSTANTS, POOL_SUB_SLOT_ITERS=case.sub_slot_iters)
+    farmer._private_keys = case.farmer_private_keys
+    farmer.authentication_keys = case.authentication_keys
+    farmer.sps[case.sp_hash] = [sp]
+    farmer.pool_state[p2_singleton_puzzle_hash] = {
+        "p2_singleton_puzzle_hash": p2_singleton_puzzle_hash.hex(),
+        "points_found_since_start": 0,
+        "points_found_24h": [],
+        "points_acknowledged_since_start": 0,
+        "points_acknowledged_24h": [],
+        "next_farmer_update": 0,
+        "next_pool_info_update": 0,
+        "current_points": 0,
+        "current_difficulty": case.pool_difficulty,
+        "pool_errors_24h": [],
+        "valid_partials_since_start": 0,
+        "valid_partials_24h": [],
+        "invalid_partials_since_start": 0,
+        "invalid_partials_24h": [],
+        "insufficient_partials_since_start": 0,
+        "insufficient_partials_24h": [],
+        "stale_partials_since_start": 0,
+        "stale_partials_24h": [],
+        "missing_partials_since_start": 0,
+        "missing_partials_24h": [],
+        "authentication_token_timeout": case.authentication_token_timeout,
+        "plot_count": 0,
+        "pool_config": case.pool_config,
+    }
+    return sp, pos, new_pos
+
+
+def override_pool_state(overrides: Dict[str, Any]) -> Dict[str, Any]:
+    pool_state = {
+        "points_found_since_start": 0,
+        # Original item format here is (timestamp, value) but we'll ignore timestamp part
+        # so every `xxx_24h` item in this dict will be List[Any].
+        "points_found_24h": [],
+        "points_acknowledged_since_start": 0,
+        "points_acknowledged_24h": [],
+        "pool_errors_24h": [],
+        "valid_partials_since_start": 0,
+        "valid_partials_24h": [],
+        "invalid_partials_since_start": 0,
+        "invalid_partials_24h": [],
+        "insufficient_partials_since_start": 0,
+        "insufficient_partials_24h": [],
+        "stale_partials_since_start": 0,
+        "stale_partials_24h": [],
+        "missing_partials_since_start": 0,
+        "missing_partials_24h": [],
+    }
+    for key, value in overrides.items():
+        pool_state[key] = value
+    return pool_state
+
+
+class PoolStateCase(DataCase):
```

Comment:
I would leverage dataclasses for this, please see `ProofOfSpaceCase` used in `test_verify_and_get_quality_string` as an example of how to go about it.